haps/container.py

```diff
@@ -277,6 +280,47 @@ def __set_name__(self, owner: Type, name: str) -> None:
             raise TypeError('No annotation for Inject')
 
 
+def _evaluate_runtime_type_hints(
+        fun: Callable, parameters: MappingProxyType) -> MappingProxyType:
+    """
+    Utility function which tries to evaluate runtime type hints of the
+    parameters using the locals of the passed function.
+
+    :param fun: function which locals should be used when evaluating
+    :param parameters: mapping of the parameter name to the parameter
+    :return: unordered mapping of the parameter name to the evaluated type
+    """
+    runtime_type_parameters = {}
+    for name, parameter in parameters.items():
+        type_ = parameter.annotation
+        if name == 'self':
+            continue
+
+        if isinstance(type_, str):
+            runtime_type_parameters[name] = parameter
+
+    if not runtime_type_parameters:
+        return MappingProxyType({
+            name: parameter.annotation
+            for name, parameter in parameters.items()
+        })
+
+    evaluated_parameter_types = {}
+
+    # Assuming that this is called directly in the inject decorator
+    inject_outer_frame = next(islice(traceback.walk_stack(None), 2, None))
+    type_hints = get_type_hints(fun, localns=inject_outer_frame[0].f_locals)
```

Comment:
That's a kinda nasty hack, don't you think? Also, you cannot assume that these types will be available two frames above - you can apply `inject` decorator anywhere in your code.

Reply:
Maybe there's a better way to check these types in a runtime?

Reply:
Also, I believe the argument name is  it's `locals`, not `localns` :smile: